api/src/components/publication/controller.ts

```diff
@@ -229,7 +221,9 @@ export const update = async (
             });
         }
 
-        const updatedPublication = await publicationService.update(event.pathParameters.id, event.body);
+        await publicationService.update(event.pathParameters.id, event.body);
```

Comment:
This is the part where it starts to get confusing, because we call a Publication service that is supposed to update a Publication, but we update a PublicationVersion instead, which is a different entity into our DB. All the checks above, are done against the publication, but the publication itself doesn't have any of those fields beside the "type". I wonder how we can improve these controllers and make it more clear what we are checking and what we are updating. I think the "update" controller and service should be renamed to "updateLatestVersion" so we know exactly what they are doing and all those checks above we can split them as follows:

const publication = await publicationService.get(event.pathParameters.id);
const latestVersion = publication.versions[0]; // get service includes latest version by default
// do all the necessary checks using "publication" and "latestVersion" accordingly

Reply:
I've pushed a commit just now that hopefully improves this, I agree that it's a clearer way of showing what exactly we're dealing with. I've not fully updated references because I am going to address those next following the other comments and make them deal directly with versions (like coauthors do)